### ISSUE TITLE
RNA fusions parsing issue?

### ISSUE THREAD

**Issue by northwestwitch** [Clinical-Genomics/scout#5148]
From a ticket in our support system:

They see 3 fusion variants hitting the genes KMT2A::AFF1 in Fusion Inspector:

<img width="1609" alt="image" src="https://github.com/user-attachments/assets/8f085a35-b21c-4eb7-a348-c074f324f54e" />

But only 2 if they find for gene KMT2A in scout. This is [the case](https://scout.scilifelab.se/cust110/F0066433-RNAseq/fusion/variants): 

<img width="1669" alt="image" src="https://github.com/user-attachments/assets/5c472726-8a52-42db-a752-01a06e07aa3b" />

I believe the problem might be in our parsing of the third variant. Check what you get if you look for the other fusion partner instead (AFF1):

<img width="1108" alt="image" src="https://github.com/user-attachments/assets/35f6bed7-1f2d-407f-b7c3-97da4846e5f0" />

**Comment by northwestwitch** [comment]
The VCF looks OK:

```
11    118482495    .    N    N[4:87084120[    .    PASS    SVTYPE=BND;CHRA=11;CHRB=4;GENEA=KMT2A;GENEB=AFF1;POSA=118482495;POSB=87084120;ORIENTATION=+,+;FOUND_DB=COSMIC,Mitelman,FusionGDB2;FOUND_IN=arriba,starfusion,fusioncatcher;TOOL_HITS=3;SCORE=0.688;FRAME_STATUS=INFRAME;TRANSCRIPT_ID_A=ENST00000531904;TRANSCRIPT_ID_B=ENST00000395146;TRANSCRIPT_VERSION_A=6;TRANSCRIPT_VERSION_B=8;HGNC_ID_A=7132;HGNC_ID_B=7135;EXON_NUMBER_A=9;EXON_NUMBER_B=5;ANNOTATIONS=Mitelman,ChimerKB,ChimerPub,Cosmic,ChimerSeq,CCLE_StarF2019,INTERCHROMOSOMAL[11--4]    GT:DV:RV:FFPM    ./1:35:4:0.4128
11    118482092    .    N    N[4:87084120[    .    PASS    SVTYPE=BND;CHRA=11;CHRB=4;GENEA=KMT2A;GENEB=AFF1;POSA=118482092;POSB=87084120;ORIENTATION=+,+;FOUND_DB=COSMIC,Mitelman,FusionGDB2;FOUND_IN=arriba,starfusion,fusioncatcher;TOOL_HITS=3;SCORE=0.688;FRAME_STATUS=FRAMESHIFT;TRANSCRIPT_ID_A=ENST00000531904;TRANSCRIPT_ID_B=ENST00000395146;TRANSCRIPT_VERSION_A=6;TRANSCRIPT_VERSION_B=8;HGNC_ID_A=7132;HGNC_ID_B=7135;EXON_NUMBER_A=8;EXON_NUMBER_B=5;ANNOTATIONS=Mitelman,ChimerKB,ChimerPub,Cosmic,ChimerSeq,CCLE_StarF2019,INTERCHROMOSOMAL[11--4]    GT:DV:RV:FFPM    ./1:3:1:0.0326
11    118482495    .    N    N[4:87084123[    .    PASS    SVTYPE=BND;CHRA=11;CHRB=4;GENEA=KMT2A;GENEB=AFF1;POSA=118482495;POSB=87084123;ORIENTATION=+,+;FOUND_DB=COSMIC,Mitelman,FusionGDB2;FOUND_IN=arriba,starfusion,fusioncatcher;TOOL_HITS=3;SCORE=0.688;FRAME_STATUS=INFRAME;TRANSCRIPT_ID_A=ENST00000531904;TRANSCRIPT_ID_B=ENST00000395146;TRANSCRIPT_VERSION_A=6;TRANSCRIPT_VERSION_B=8;HGNC_ID_A=7132;HGNC_ID_B=7135;EXON_NUMBER_A=9;EXON_NUMBER_B=5;ANNOTATIONS=Mitelman,ChimerKB,ChimerPub,Cosmic,ChimerSeq,CCLE_StarF2019,INTERCHROMOSOMAL[11--4]    GT:DV:RV:FFPM    ./1:2:2:0.0224
```

**Comment by northwestwitch** [comment]
Inspecting the VCF I found this variant, which is also mapping to the exact coordinates of the missing variant, the one with gene KMT2A:

Since this variant with gene A AP001267.5 instead of KMT2A occurs before in the VCF, it gets loaded instead of the latter because of how scout works (same problem as the variant in CALR gene loaded only once if found in SNVs and SVs). So it's not really a bug but a design thing (or limitation), I guess.

`11	118482495	.	N	N[4:87084120[	.	PASS	SVTYPE=BND;CHRA=11;CHRB=4;GENEA=AP001267.5;GENEB=AFF1;POSA=118482495;POSB=87084120;ORIENTATION=nan,nan;FOUND_DB=nan;FOUND_IN=arriba,fusioncatcher;TOOL_HITS=2;SCORE=0.125;FRAME_STATUS=nan;TRANSCRIPT_ID_A=nan;TRANSCRIPT_ID_B=nan;TRANSCRIPT_VERSION_A=0;TRANSCRIPT_VERSION_B=0;HGNC_ID_A=0;HGNC_ID_B=7135;EXON_NUMBER_A=0;EXON_NUMBER_B=0;ANNOTATIONS=nan	GT:DV:RV:FFPM	./1:0:0:0.0`

Of course one might argue that this is not the intended behaviour. I'm not really sure how to fix thing. 🤔

**Comment by dnil** [comment]
Yes, this is a case of a poorly merged VCF. Two variants, of the same type, that have the same start and end coordinates should be represented on only one row. They should have the same VCF annotation, possibly merged between between the two variant annotations above if the "AP001267.5" is indeed a proper transcript, and not just a poorly described isoform of KMT2A. 

We also have a genuine collision issue in scout, which is normally between multiple files for the same case, in the overlap zone where with multiple callers of different kind can find the the same "simple ids" (chrom, pos, ref, alt) are all different. 

Anyway, this is apparently now fixed in https://github.com/FusionInspector/FusionInspector/pull/82 (well done!) so we should be able to close.

**Comment by dnil** [comment]
Further discussed in https://github.com/Clinical-Genomics/MTP-RNAFUSION/issues/76 and https://github.com/Clinical-Genomics/MTP-RNAFUSION/issues/101. 🚀
